core/src/test/java/org/jdbi/v3/core/statement/TestColonPrefixSqlParser.java

```diff
@@ -33,77 +32,146 @@ public void setUp() {
+                .append("~* ")
+                .appendNamedParameter("boo")
+                .append(" ':nope' _%&^& *@ ")
+                .appendNamedParameter("id")
+                .build());
     }
 
     @Test
     public void testNumbers() {
-        ParsedSql parsed = parser.parse(":bo0 ':nope' _%&^& *@ :id", ctx);
-        assertThat(parsed.getSql()).isEqualTo("? ':nope' _%&^& *@ ?");
+        assertThat(parser.parse(":bo0 ':nope' _%&^& *@ :id", ctx))
+            .isEqualTo(ParsedSql.builder()
+                .appendNamedParameter("bo0")
+                .append(" ':nope' _%&^& *@ ")
+                .appendNamedParameter("id")
+                .build());
     }
 
     @Test
     public void testDollarSignOkay() {
-        ParsedSql parsed = parser.parse("select * from v$session", ctx);
-        assertThat(parsed.getSql()).isEqualTo("select * from v$session");
+        String sql = "select * from v$session";
+        assertThat(parser.parse(sql, ctx))
+            .isEqualTo(ParsedSql.builder().append(sql).build());
     }
 
     @Test
     public void testHashInColumnNameOkay() {
-        ParsedSql parsed = parser.parse("select column# from thetable where id = :id", ctx);
-       assertThat(parsed.getSql()).isEqualTo("select column# from thetable where id = ?");
+        assertThat(parser.parse("select column# from thetable where id = :id", ctx))
+            .isEqualTo(ParsedSql.builder()
+                .append("select column# from thetable where id = ")
+                .appendNamedParameter("id")
+                .build());
     }
 
     @Test
     public void testBacktickOkay() {
-        ParsedSql parsed = parser.parse("select * from `v$session", ctx);
-        assertThat(parsed.getSql()).isEqualTo("select * from `v$session");
+        String sql = "select * from `v$session";
+        assertThat(parser.parse(sql, ctx))
+            .isEqualTo(ParsedSql.builder().append(sql).build());
     }
 
     @Test
     public void testDoubleColon() {
         final String doubleColon = "select 1::int";
-        ParsedSql parsed = parser.parse(doubleColon, ctx);
-        assertThat(parsed.getSql()).isEqualTo(doubleColon);
-    }
-
-    @Test
-    public void testBailsOutOnInvalidInput() {
-        assertThatThrownBy(() -> parser.parse("select * from something\n where id = :\u0087\u008e\u0092\u0097\u009c", ctx).getSql())
-            .isInstanceOf(UnableToCreateStatementException.class);
+        assertThat(parser.parse(doubleColon, ctx))
+            .isEqualTo(ParsedSql.builder().append(doubleColon).build());
     }
 
     @Test
-    public void testSubstitutesDefinedAttributes() {
-        String sql = "select foo from bar where foo = :someValue";
-        ParsedSql parsed = parser.parse(sql, ctx);
-        assertThat(parsed.getSql()).isEqualTo("select foo from bar where foo = ?");
+    public void testColonWithoutNameNotAParameter() {
+        String sql = "select * from something\n where id = :\u0087\u008e\u0092\u0097\u009c";
```

Comment:
could we have this magic unicode string explained with a comment or `describedAs`? I don't see how this is testing what the method name says because it looks like a named parameter to me.